velox/type/tests/DecimalTest.cpp

```diff
@@ -23,25 +23,29 @@ namespace facebook::velox {
 
 namespace {
 
-template <typename T>
-void assertRescaleDouble(double value, const TypePtr& type, T expectedValue) {
+template <typename TInput, typename TOutput>
+void assertRescaleFloatingPoint(
+    TInput value,
+    const TypePtr& type,
+    TOutput expectedValue) {
   const auto [precision, scale] = getDecimalPrecisionScale(*type);
-  T actualValue;
-  const auto status =
-      DecimalUtil::rescaleDouble<T>(value, precision, scale, actualValue);
+  TOutput actualValue;
+  const auto status = DecimalUtil::rescaleFloatingPoint<TInput, TOutput>(
+      value, precision, scale, actualValue);
   ASSERT_TRUE(status.ok());
   ASSERT_EQ(actualValue, expectedValue);
 }
 
-template <typename T>
-void assertRescaleDoubleFail(
-    double value,
+template <typename TInput, typename TOutput>
+void assertRescaleFloatingPointFail(
+    TInput value,
     const TypePtr& type,
     const std::string& expectedErrorMessage) {
   const auto [precision, scale] = getDecimalPrecisionScale(*type);
-  T actualValue;
+  TOutput actualValue;
   VELOX_ASSERT_ERROR_STATUS(
-      DecimalUtil::rescaleDouble<T>(value, precision, scale, actualValue),
+      (DecimalUtil::rescaleFloatingPoint<TInput, TOutput>(
```

Comment:
nit: drop extra () around DecimalUtil::rescaleFloatingPoint

Reply:
This is to fix a compilation error. The macro does not recognise the `<` and `>` brackets as special character, but see the `,` in the middle as an argument separator, so it throws `to many arguments` error.

Reply:
I see. Thank you for explaining.